### PR TITLE
Add Popup and PopupMenu

### PR DESCRIPTION
**Pull request by sliptype**
This PR:

* Adds `Popup` and `PopupMenu`
* Adds & configures the `import/order` eslint rule
* Enables the `import/no-default-export` eslint rule
* Nests all components under `components`
* Renames `index.tsx` files to `index.ts`

(This probably should have been 2 PRs but I tried splitting it up retroactively and ran into issues)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,6 +73,7 @@
     "eslint-config-prettier": "^6.11.0",
```
Comment (raymondjacobson) [inline package.json:73]: just checking which npm version you have for the package-lock changes

File: src/components/Popup/types.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  onAfterClose?: () => void
+  /**
+   * The position of the Popup relative to the trigger
+   */
+  position?: Position
+  /**
+   * A title displayed at the top of the Popup (only visible when the header is enabled)
+   */
+  title?: string
+  wrapperClassName?: string
```
Comment (raymondjacobson) [inline src/components/Popup/types.ts:53]: maybe worth a comment on what the wrapper is

File: src/components/PopupMenu/types.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { PopupProps } from '../Popup'
+
+export type PopupMenuProps = {
+  /**
+   * The items to display in the menu
+   */
+  items: PopupMenuItem[]
+
```
Comment (raymondjacobson) [inline src/components/PopupMenu/types.tsx:8]: nit we should be consistent about newline spacing b/w prop fields

**Review by raymondjacobson — APPROVED** [review]
Looks great! Would you mind holding off on the merge here until I sort out what happened w/ the scrubber in the latest version change?

File: package.json (hunk 1 above)
Comment (sliptype) [reply package.json:73]: I'm on `6.14.13` now, these changes are going back from `lockFileVersion: 2` to `lockFileVersion: 1`

File: src/components/Popup/types.ts (hunk 2 above)
Comment (sliptype) [reply src/components/Popup/types.ts:57]: For sure, added!

File: src/components/PopupMenu/types.tsx (hunk 3 above)
Comment (sliptype) [reply src/components/PopupMenu/types.tsx]: Fixed!

**Pull request merged**